src/parser/ts/index.ts

```diff
@@ -217,24 +217,30 @@ export const getIndicesOfMultiaryFunctionArgs = (route: STSQuery): number[] => {
   return responseArr;
 };
 
-export const getServerFilters = (
-  route: STSQuery
-): StringMap[] => {
-  const responseArr: StringMap[] = [];
-  const filter = (_, __, parent) => isServerFilter(parent);
+export const extractFilters = (
+  route: STSQuery,
+  condition: (_, __, parent) => boolean
+): unknown[][] => {
+  const extracted: unknown[][] = [];
   walk(route, obj => {
     if (isSTSReference(obj) && !hasIdsFilter(obj)) {
-      const serverQuery = filterDeep(filterDeep(obj.query, filter), filter);
-      const flatFilters = flattenServerQueryFilters(serverQuery || []);
-      responseArr.push(flatFilters);
+      extracted.push(filterDeep(obj.query, condition));
     }
   });
-  return responseArr;
+  return extracted;
+};
+
+export const getServerFilters = (
+  route: STSQuery
+): StringMap[] => {
+  const filter = (_, __, parent) => isServerFilter(parent) || !isSTSFilter(parent);
```

Comment:
Does `isServerFilter(parent)` not enough?

Reply:
no. it skips `ts{prop=[...arr]}`, as parent just a regular array, not the object

Reply:
got it